### PR TITLE
Orphan/Adopt Enhancement

### PR DESCRIPTION
**Pull request by paulcarlton-ww**

### CODE REVIEW

**Comment by paulcarlton-ww** [comment]
This supersedes https://github.com/fidelity/kraan/pull/177 which I closed because I messed up the rebase
Outstanding work is

Docs
Unit tests
Integration Tests

**Comment by paulcarlton-ww** [comment]
Also may need https://github.com/fidelity/kraan/pull/167/ to cater for empty layers

**Comment by saada** [comment]
If you need a dev version of helm chart published, you'll need to `git rebase -i origin/master`

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -91,7 +91,20 @@ jobs:
...
+          kubectl patch gitrepository addons-config -n gotk-system -p "{\"spec\": {\"ref\": {\"branch\": \"$GITHUB_HEAD_REF\"}}}" --type=merge
+          kubectl apply -f ./testdata/addons-orphan/addons-repo.yaml
+          kubectl apply -f ./testdata/addons-orphan/addons.yaml
 
+          echo "waiting 5m for test to pass..."
+          kubectl wait al --for=condition=Deployed --all --timeout=5m || (kubectl get po -A && false)
+          sleep 1
+          kubectl get al
+          kubectl get helmreleases.helm.toolkit.fluxcd.io -A  -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.namespace}{"\t"}{.metadata.labels}{"\t"}{.metadata.ownerReferences[0].name}{"\t"}{"\n"}{end}'
+          kubectl get addonslayers.kraan.io -A  -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{range .status.resources[*]}{.name}{"\t"}{.namespace}{"\n"}{end}{"\n"}{end}'
```
Comment (saada) [inline .github/workflows/main.yaml:107]: This prints the output but requires a beautiful human being to verify its correctness. I suggest adding a grep or some conditional test that will pass/fail. Unless waiting for ALs to be deployed is enough validation

**Comment by saada** [comment]
Code looks good but I can't wrap my head around how to orphan and adopt in practice on a running cluster. Maybe a guide such as "How do I move a HR with no downtime from one layer to another?"

File: .github/workflows/main.yaml (hunk 1 above)
Comment (paulcarlton-ww) [reply .github/workflows/main.yaml:107]: Waiting for deployment should be enough, I added this to prove to myself it works, will remove it or do as you suggest and parse output

Comment (paulcarlton-ww) [reply .github/workflows/main.yaml]: removed

**Review by saada — APPROVED** [review]
LGTM

**Pull request merged**